rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -18,19 +21,48 @@
 from rotkehlchen.types import ChecksumEvmAddress
 from rotkehlchen.user_messages import MessagesAggregator
 
+from .constants import CPT_LIQUITY
 
 if TYPE_CHECKING:
     from rotkehlchen.assets.asset import Asset
     from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
     from rotkehlchen.chain.evm.contracts import EvmContract
     from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
 
 MIN_COLL_RATE = '1.1'
 
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
+QUERY_STAKING_EVENTS = """
+WHERE event_identifier IN
+(SELECT A.event_identifier FROM history_events AS A JOIN history_events AS B ON A.event_identifier = B.event_identifier WHERE
```

Comment:
Why to join history_events to history_events?

Reply:
Because there are multiple rewards collected in the same transaction but we don't have an easy way to differenciate what are rewards from the stability pool and what are rewards from staking. The only difference is that you are rewarded LQTY from the stability pool and LUSD from staking. This is why I search for this special assets and then get the events in the transactions containing them

Reply:
Hmmm okay. Looks weird, but probably makes sense